Polygon_mesh_processing/doc/Polygon_mesh_processing/Concepts/PMPCorefinementVisitor.h

```diff
@@ -117,4 +117,61 @@ typedef unspecified_type halfedge_descriptor;
   void after_vertex_copy(vertex_descriptor v_src, const Triangle_mesh& tm_src,
                          vertex_descriptor v_tgt, const Triangle_mesh& tm_tgt);
 /// @}
+
+/// @name Functions used by corefine() for progress tracking
+/// @{
+  /// called before starting to detect intersections between faces from a mesh and edges of the other
+  void start_filtering_intersections();
+  /// called during detection of intersections between faces from a mesh and edges of the other.
+  /// `d` is a double value in `[0,1]` that is increasing with the number of calls. The closer
+  /// `d`is to `1`, the closer the intersection detection is to completion.
+  void progress_filtering_intersections(double d);
+  /// called after detection of intersections between faces from a mesh and edges of the other
+  void end_filtering_intersections();
+
+  /// called before processing intersections between the `n` pairs of coplanar faces
+  void start_handling_intersection_of_coplanar_faces(std::size_t n);
+  /// called each time a pair of coplanar face is processed
+  void intersection_of_coplanar_faces_step() const;
+  /// called after processing all intersections between coplanar faces
+  void end_handling_intersection_of_coplanar_faces() const;
+
+  /// called before processing intersections between edges and faces of two meshes (called twice).
+  /// `n` is the number of edges possibly intersecting faces that will be processed.
+  void start_handling_edge_face_intersections(std::size_t n);
+  /// called each time an edge is processed
+  void edge_face_intersections_step();
+  /// called after having processed edge-face intersections between two meshes
+  void end_handling_edge_face_intersections();
+
+  /// called before triangulating the `n` split faces
+  void start_triangulating_faces(std::size_t n);
+  /// called when triangulating one split face
+  void triangulating_faces_step();
+  /// called after the triangulation of the split faces
+  void end_triangulating_faces();
+/// @}
+
+/// @name Functions used by Boolean operations functions using corefinement for progress tracking.
+/// These functions are not needed if only `corefine()` is called.
+/// called before computing the output of the Boolean operations after corefinement
+/// @{
+  void start_building_output();
+  /// called when the outputs of the Boolean operations is computed
```

Comment:
```suggestion
  /// called when the outputs of the Boolean operations are computed
```

Reply:
I was wondering if it should not be operation  instead of operations

Reply:
I've fixed start with output but forgot about end